witmotion_imu:
    port: ttyUSB0
    baud_rate: 115200 # baud
    polling_interval: 10 # ms
    restart_service_name: /restart_imu
    imu_publisher:
        topic_name: /imu
        frame_id: imu
        use_native_orientation: false
        measurements:
            acceleration:
                enabled: true
                covariance: [ 0.0112,    0,       0,
                              0,         0.0081,  0,
                              0,         0,       0.0128 ]
            angular_velocity:
                enabled: false
                covariance: [ 0.1199,    0,       0,
                              0,         0.5753,  0,
                              0,         0,       0.0267 ]
            orientation:
                enabled: true
                covariance: [ 0.0457,    0,       0,
                              0,         0.0655,  0,
                              0,         0,       0      ]
    temperature_publisher:
        enabled: false
        topic_name: /temperature
        frame_id: base_link
        from_message: magnetometer # acceleration, angular_vel, orientation, magnetometer
        variance: 0.02683
        coefficient: 1.0 # Linear calibration parameters: coefficient
        addition: 0.0 # and addendum
    magnetometer_publisher:
        enabled: false
        topic_name: /magnetometer
        frame_id: imu
        coefficient: 0.00000001 # Linear calibration parameters: coefficient
        addition: 0.0 # and addendum
        covariance: [ 0.000000187123,   0,              0,
                      0,                0.000000105373, 0,
                      0,                0,              0.000000165816  ]
    barometer_publisher:
        enabled: false
        topic_name: /barometer
        frame_id: base_link
        variance: 0.001
        coefficient: 1.0 # Linear calibration parameters: coefficient
        addition: 0.0 # and addendum
    altimeter_publisher:
        enabled: false
        topic_name: /altitude
        coefficient: 1.0 # Linear calibration parameters: coefficient
        addition: 0.0 # and addendum
    orientation_publisher:
        enabled: false
        topic_name: /orientation
    gps_publisher:
        enabled: false
        navsat_fix_frame_id: world
        navsat_fix_topic_name: /gps
        navsat_altitude_topic_name: /gps_altitude
        navsat_satellites_topic_name: /gps_satellites
        navsat_variance_topic_name: /gps_variance
        ground_speed_topic_name: /gps_ground_speed
    rtc_publisher:
        enabled: false
        topic_name: /witmotion_clock

